modules/charon-core/src/test/java/org/wso2/charon3/core/protocol/endpoints/UserResourceManagerTest.java

```diff
@@ -115,89 +364,865 @@ public void testGetSuccess(String id, String attributes,
+    }
+
+    @Test
+    public void testListWithPOSTNotImplementedException()
+            throws NotImplementedException, BadRequestException, CharonException {
+
+        abstractResourceManager.when(() -> AbstractResourceManager.getResourceEndpointURL(SCIMConstants.USER_ENDPOINT))
+                .thenReturn(SCIM2_USER_ENDPOINT + "/.search");
+        abstractResourceManager.when(()
+                -> AbstractResourceManager.encodeSCIMException(any(NotImplementedException.class)))
+                .thenReturn(getEncodeSCIMExceptionObject(new NotImplementedException()));
+        Mockito.when(userManager.listUsersWithPost(any(SearchRequest.class),
+                anyMap())).thenThrow(new NotImplementedException());
+        SCIMResponse scimResponse = userResourceManager.listWithPOST(RESOURCE_STRING, userManager);
+        Assert.assertEquals(scimResponse.getResponseStatus(), ResponseCodeConstants.CODE_NOT_IMPLEMENTED);
+    }
+
+    @Test
+    public void testListWithPOSTBadRequestException()
+            throws NotImplementedException, BadRequestException, CharonException {
+
+        abstractResourceManager.when(() -> AbstractResourceManager.getResourceEndpointURL(SCIMConstants.USER_ENDPOINT))
+                .thenReturn(SCIM2_USER_ENDPOINT + "/.search");
+        abstractResourceManager.when(()
+                -> AbstractResourceManager.encodeSCIMException(any(BadRequestException.class)))
+                .thenReturn(getEncodeSCIMExceptionObject(new BadRequestException()));
+        Mockito.when(userManager.listUsersWithPost(any(SearchRequest.class),
+                anyMap())).thenThrow(new BadRequestException());
+        SCIMResponse scimResponse = userResourceManager.listWithPOST(RESOURCE_STRING, userManager);
+        Assert.assertEquals(scimResponse.getResponseStatus(), ResponseCodeConstants.CODE_BAD_REQUEST);
+    }
+
+    @DataProvider(name = "dataForUpdateWithPATCH")
+    public Object[][] dataToUpdateWithPATCH() throws BadRequestException, CharonException, InternalErrorException {
+
+        SCIMResourceTypeSchema schema = SCIMResourceSchemaManager.getInstance().getUserResourceSchema();
+        JSONDecoder decoder = new JSONDecoder();
+        User userOld = decoder.decodeResource(NEW_USER_SCIM_OBJECT_STRING_FOR_PATCH, schema, new User());
+        String id = userOld.getId();
+        User userNew = decoder.decodeResource(NEW_USER_SCIM_OBJECT_STRING_FOR_PATCH_UPDATE, schema, new User());
+        return new Object[][]{
+                {id, NEW_USER_SCIM_OBJECT_STRING_FOR_PATCH_UPDATE, "userName", null, userNew, userOld}
+        };
+    }
+
+    @Test(dataProvider = "dataForUpdateWithPATCH")
+    public void testUpdateWithPATCH(String existingId, String scimObjectString,
+                                    String attributes, String excludeAttributes, Object scimNewUserObject,
+                                    Object scimOldUserObject)
```

Comment:
check if this can be merged into 2 lines

Reply:
Fixed in commit 504369f8bdb4f024c63ae5d78f16b04decbedc11

Reply:
public void testUpdateWithPATCH(String existingId, String scimObjectString, String attributes, 
                                    String excludeAttributes, Object scimNewUserObject, Object scimOldUserObject)

Reply:
Fixed in  commit af77bf94a86e4b649d26ce314d5408b2fde6750d